### PR TITLE
Closes #568: Report citation matching counters

### LINKED ISSUE #568: Numbers related to citation matching results
matched citations
    - total (do not confuse with number of publications for which citations were matched, this could be measured with different counter)
    - number of extracted citation texts (this could be also measured at importing phase)
    - number of publications with extracted citations texts (at least one bibref)
    - total matched citations by fuzzy algorithm
    - total matched citations by direct algorithm

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectReporter.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import pl.edu.icm.sparkutils.avro.SparkAvroSaver;
+
+/**
+ * Reporter of citation matching direct job counters.<br/>
+ * It calculates citation matching direct counters and saves them
+ * as {@link ReportParam} datastore.
+ * 
+ * @author madryk
+ */
+public class CitationMatchingDirectReporter {
```
Comment (marekhorst) [inline iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectReporter.java]: I would add `Counter` phrase in this name to make it straight-forward in terms of functionality: `CitationMatchingDirectCounterReporter`. This may seem to be too wordy though... Since it is already in `*.citationmatching.direct.*` package we could make `CounterReporter` out of it. What do you think?

File: iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectReporter.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    
+    
+    private SparkAvroSaver avroSaver = new SparkAvroSaver();
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Calculates and saves citation matching counters using citations rdd
+     * under outputReportPath.
```
Comment (marekhorst) [inline iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectReporter.java]: I would rephrase it, currently it seems citations rdd is under `outputReportPath` ;)

File: iis-wf/iis-wf-citationmatching/src/main/java/eu/dnetlib/iis/wf/citationmatching/CitationMatchingReporter.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import pl.edu.icm.sparkutils.avro.SparkAvroSaver;
+
+/**
+ * Reporter of citation matching job counters.<br/>
+ * It calculates citation matching counters and saves them
+ * as {@link ReportParam} datastore.
+ * 
+ * @author madryk
+ */
+public class CitationMatchingReporter {
```
Comment (marekhorst) [inline iis-wf/iis-wf-citationmatching/src/main/java/eu/dnetlib/iis/wf/citationmatching/CitationMatchingReporter.java]: My comment related to `CitationMatchingDirectReporter` class name is applicable here as well. Since reporter is counter based it will be easier to understand what it does having `Counter` in class name.

File: iis-wf/iis-wf-citationmatching/src/main/java/eu/dnetlib/iis/wf/citationmatching/CitationMatchingReporter.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    private String reportPath;
+    
+    private JavaSparkContext sparkContext;
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Calculates and saves citation matching counters using matchedCitations rdd
+     * under {@link #setReportPath(String)}
```
Comment (marekhorst) [inline iis-wf/iis-wf-citationmatching/src/main/java/eu/dnetlib/iis/wf/citationmatching/CitationMatchingReporter.java]: Rephrase? `Calculates citation matching counters using matchedCitations rdd ans saves them under {@link #setReportPath(String)}`

File: iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectReporter.java (hunk 1 above)
Comment (madryk) [inline iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectReporter.java]: I renamed it to `CitationMatchingDirectCounterReporter`.
Not a big fan of option to shorten this to `CounterReporter`. It would be easy to mistake this with class in fuzzy citation matching module or find that there is the only one general `CounterReporter` that could be used for every module that want to report something (usually very little attention is paid to reading import headers in java).

File: iis-wf/iis-wf-citationmatching/src/main/java/eu/dnetlib/iis/wf/citationmatching/CitationMatchingReporter.java (hunk 3 above)
Comment (madryk) [inline iis-wf/iis-wf-citationmatching/src/main/java/eu/dnetlib/iis/wf/citationmatching/CitationMatchingReporter.java]: I renamed it to `CitationMatchingCounterReporter`

File: iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectCounterReporter.java [hunk 5]
```diff
@@ -29,8 +29,8 @@
     //------------------------ LOGIC --------------------------
     
     /**
-     * Calculates and saves citation matching counters using citations rdd
-     * under outputReportPath.
+     * Calculates citation matching counters using citations rdd
+     * and saved them under outputReportPath.
```
Comment (marekhorst) [inline iis-wf/iis-wf-citationmatching-direct/src/main/java/eu/dnetlib/iis/wf/citationmatching/direct/service/CitationMatchingDirectCounterReporter.java]: Typo. saved->saves.

**Comment by marekhorst** [comment]
One typo, you can fix it and merge changes with cdh5.
